packages/authentication/test/unit/strategy-adapter.test.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {ShimRequest} from '../..';
+import {StrategyAdapter} from '../..';
+import {ParsedRequest} from '@loopback/core';
+
+const passportStrategy = require('passport-strategy');
+
+describe('Strategy Adapter', () => {
+  const mockUser: User = {id: 'mock-user', role: 'mock-role'};
+  interface User {
+    id: string;
+    role: string;
+  }
+
+  describe('authenticate()', () => {
+    it('calls the authenticate method of the strategy', () => {
+      const strategy = new passportStrategy();
+      let calledFlag = false;
+      // TODO: (as suggested by @bajtos) use sinon spy
+      strategy.authenticate = function() {
+        calledFlag = true;
+      };
+      const adapter = new StrategyAdapter(strategy);
+      const request = <ParsedRequest> {};
+      // tslint:disable-next-line:no-floating-promises
+      const user: Object = adapter.authenticate(request);
```

Comment:
> it does not run synchronously. authenticate in mock strategy calls process.nexttick

I think you need to `await` the result of `adapter.authenticate(request)` then, similarly to what you are doing in the next test.